### PR TITLE
Adding TypeScript to "Getting Started" 

### PR DESCRIPTION
**Pull request by kaiwalyakoparkar**
Ref #1800 and #1888
@svrnm @chalin @cartermp kindly take a look

### CODE REVIEW

**Comment by cartermp** [comment]
Looks like you were right about the commits. Thanks for sticking it out like this. Looks like there's failures with respect to the main branch changes now, but once those are taken care of, I can approve.

**Comment by kaiwalyakoparkar** [comment]
> Looks like you were right about the commits. Thanks for sticking it out like this. Looks like there's failures with respect to the main branch changes now, but once those are taken care of, I can approve.

Thanks for the information. It's been fixed 👍

**Review by cartermp — APPROVED** [review]

**Comment by cartermp** [comment]
@open-telemetry/javascript-approvers PTAL!

**Comment by svrnm** [comment]
thanks for figthing this true! TY

**Review by svrnm — APPROVED** [review]

File: content/en/docs/instrumentation/js/getting-started/nodejs.md
Comment (weyert) [inline content/en/docs/instrumentation/js/getting-started/nodejs.md:154]: I don't think we need to introduce a TypeScript  dependency int his Getting Start. As far as I can see the `: any` is the only TypeScript typing in this code snippet. I think using vanilla Node/JavaScript makes it easier to get things going

File: content/en/docs/instrumentation/js/getting-started/nodejs.md
Comment (svrnm) [inline content/en/docs/instrumentation/js/getting-started/nodejs.md:2]:
```suggestion
title: Node.js
```

File: content/en/docs/instrumentation/js/getting-started/nodejs.md
Comment (chalin) [inline content/en/docs/instrumentation/js/getting-started/nodejs.md:6]:
```suggestion
This guide will show you how to get started with tracing in Node.js.
```

WDYT @svrnm?

File: content/en/docs/instrumentation/js/getting-started/nodejs.md
Comment (chalin) [inline content/en/docs/instrumentation/js/getting-started/nodejs.md:12]:
```suggestion
Ensure that you have the following installed locally:

- [Node.js](https://nodejs.org/en/download/)
- [TypeScript](https://www.typescriptlang.org/download), if you will be using TypeScript.
```

File: content/en/docs/instrumentation/js/getting-started/nodejs.md
Comment (chalin) [inline content/en/docs/instrumentation/js/getting-started/nodejs.md:52]:
```suggestion
{{< ot-tab lang="ts" >}}
```

File: content/en/docs/instrumentation/js/getting-started/nodejs.md
Comment (chalin) [inline content/en/docs/instrumentation/js/getting-started/nodejs.md:68]:
```suggestion
{{< ot-tab lang="js" >}}
```

File: content/en/docs/instrumentation/js/getting-started/nodejs.md
Comment (chalin) [inline content/en/docs/instrumentation/js/getting-started/nodejs.md:107]:
```suggestion
The following dependencies are required to trace a Node.js application.
```

WDYT @svrnm?

File: content/en/docs/instrumentation/js/getting-started/nodejs.md
Comment (chalin) [reply content/en/docs/instrumentation/js/getting-started/nodejs.md:154]: Good point, and I would tend to agree.
Comments @open-telemetry/javascript-approvers @svrnm @cartermp?

**Review by chalin — COMMENTED** [review]
Thanks. See inline comments, and questions for @svrnm and other reviewers.

File: content/en/docs/instrumentation/js/getting-started/nodejs.md
Comment (svrnm) [reply content/en/docs/instrumentation/js/getting-started/nodejs.md:6]: yes, I think that makes sense, because at the end nodejs is used

File: content/en/docs/instrumentation/js/getting-started/nodejs.md
Comment (cartermp) [reply content/en/docs/instrumentation/js/getting-started/nodejs.md:154]: I think it's fine to just rely on type inference here for those code sample, but I disagree with not having TypeScript as a way to get started. There's a ton of TS developers who don't use vanilla JS anymore.

**Comment by svrnm** [comment]
Really want to have this merged soon, @kaiwalyakoparkar can you give the reviews another look?:)

**Comment by kaiwalyakoparkar** [comment]
> Really want to have this merged soon, @kaiwalyakoparkar can you give the reviews another look?:)

Sure, I will commit the required changes soon

File: content/en/docs/instrumentation/js/getting-started/browser.md
Comment (chalin) [inline content/en/docs/instrumentation/js/getting-started/browser.md:12]:
```suggestion
Ensure that you have the following installed locally:

- [Node.js](https://nodejs.org/en/download/)
```

File: content/en/docs/instrumentation/js/getting-started/nodejs.md
Comment (chalin) [reply content/en/docs/instrumentation/js/getting-started/nodejs.md:155]: So, what's the final call here? @svrnm - any thoughts/preferences?

**Review by chalin — APPROVED** [review]

File: content/en/docs/instrumentation/js/getting-started/nodejs.md
Comment (cartermp) [reply content/en/docs/instrumentation/js/getting-started/nodejs.md:155]: I vote to keep for now - we can adjust this after the fact if we need to.

(also this is for the typescript tab, not the JS tab, so there's no real harm here)

Comment (svrnm) [reply content/en/docs/instrumentation/js/getting-started/nodejs.md:155]: It was the whole purpose of this PR to have TS & JS side-by-side, I agree that the added value is not big since most code looks almost identically, but from an end user perspective it's sometimes just about "let me copy & paste that so I can get started quickly" and that's given that way.

Comment (kaiwalyakoparkar) [reply content/en/docs/instrumentation/js/getting-started/nodejs.md:155]: @svrnm so nothing to update right?

**Comment by svrnm** [comment]
@open-telemetry/javascript-approvers please give this another look :-)

File: content/en/docs/instrumentation/js/getting-started/browser.md
Comment (legendecas) [inline content/en/docs/instrumentation/js/getting-started/browser.md:67]: I don't think browsers are able to load typescript source codes directly. For both JavaScript and TypeScript projects, here the src attribute value should be `document-load.js` (the output of tsc compilation).

File: content/en/docs/instrumentation/js/getting-started/nodejs.md
Comment (legendecas) [inline content/en/docs/instrumentation/js/getting-started/nodejs.md:155]: If the explicit typing is `any`, I'd suggest leaving it for TypeScript to deduce the correct type.

```suggestion
const sdk = new opentelemetry.NodeSDK({
```

**Review by legendecas — COMMENTED** [review]
Overall LGTM % some nits.

File: content/en/docs/instrumentation/js/getting-started/browser.md
Comment (svrnm) [reply content/en/docs/instrumentation/js/getting-started/browser.md:67]: If I remember correctly parel (the build tool used in the example) will infer the type of the file from that and then do a proper compile and replacement. @kaiwalyakoparkar have you run through that example with TS and can verify, if not let's try this out.

Comment (kaiwalyakoparkar) [reply content/en/docs/instrumentation/js/getting-started/browser.md:67]: Hey @svrnm I did run these files and can confirm that the output I received is according to the docs

File: content/en/docs/instrumentation/js/getting-started/nodejs.md
Comment (kaiwalyakoparkar) [reply content/en/docs/instrumentation/js/getting-started/nodejs.md]: Oh, that makes sense. I have removed it @legendecas

File: content/en/docs/instrumentation/js/getting-started/browser.md
Comment (dyladan) [reply content/en/docs/instrumentation/js/getting-started/browser.md:67]: IMO it's quite confusing as parcel isn't mentioned until after the code samples. If you want to depend on bundler-specific behavior I would at least call it out at the top of the document that the samples will require some particular build process or dependency.

**Review by dyladan — COMMENTED** [review]
Code samples look good to me. I share @legendecas reservation about the ts extension when it isn't clear that this is enabled by build tooling

File: content/en/docs/instrumentation/js/getting-started/browser.md
Comment (svrnm) [reply content/en/docs/instrumentation/js/getting-started/browser.md:67]: @dyladan fair point! That one is on me, I wrote that tutorial initially. Let's have this in a separate (follow-up) issue

Comment (cartermp) [reply content/en/docs/instrumentation/js/getting-started/browser.md:67]: I filed https://github.com/open-telemetry/opentelemetry.io/issues/1989 for this issue

**Comment by cartermp** [comment]
I will go ahead and merge, then we can address https://github.com/open-telemetry/opentelemetry.io/issues/1989 orthogonally, since it is all that remains. Thanks all, and especially @kaiwalyakoparkar for your diligence and patience across several PRs!

**Pull request merged**
